app/models/import.rb

```diff
@@ -46,6 +46,15 @@ def update_csv!(row_idx:, col_idx:, value:)
     update! normalized_csv_str: updated_csv.to_s
   end
 
+  def add_headers_if_missing
+    return if raw_csv_str.blank?
+    first_raw = CSV.new(raw_csv_str).shift
+
+    unless header_row?(first_raw)
+      self.raw_csv_str = first_raw.join(",") + "\n" + raw_csv_str
```

Comment:
header raw is made same as the first raw. Adding a default raw like  "date,name,category,tag,amount" might confuse the user in the next configuration step

Reply:
The trade-off of this approach is that while it makes the configuration step easier for the user, it makes debugging harder. Someone looking at the database might think the import missed a row because there is no obvious header.

Reply:
Yeah, as I'm looking at this, I'm wondering if this trade-off is worth it.  When I had originally thought about a solution here I had not considered how brittle it might be to determine whether the first row is a header or a value.

I think this could get really tricky if a user provides a CSV where the transaction names and categories come first.

Would love to hear your thoughts on this, but I'm thinking we'd be better off requiring a user to provide headers for now so that our debugging can remain simple.

We could still provide a simple validation to ensure there are at least 2 headers (for our required data, amount fields), but probably not worth the hassle past that?

Reply:
>  I'm thinking we'd be better off requiring a user to provide headers for now so that our debugging can remain simple.

We would still need to determine whether the first raw is a header. Assuming banks provide CSV export in different languages/formats the only thing that could work is checking if at least one column is numeric or not.

Reply:
@tonyvince are you saying that we'd still need to determine whether the first row is a header so that we can support both types of CSVs?

My view is that we should just assume all CSVs have headers and let the user fix any issues that arise from it (similar to how we're currently doing)

Long term I can see the use case for adding more automation around this, but I don't think it's a necessary feature right now.

Reply:
ah ok. Makes sense